cvmfs/cvmfs.cc

```diff
@@ -1331,6 +1331,25 @@ static void cvmfs_getxattr(fuse_req_t req, fuse_ino_t ino, const char *name,
     attribute_value = string(VERSION) + "." + string(CVMFS_PATCH_LEVEL);
   } else if (attr == "user.pubkeys") {
     attribute_value = mount_point_->signature_mgr()->GetActivePubkeys();
+  } else if (attr == "user.repo_counters") {
+    attribute_value = mount_point_->catalog_mgr()->GetRootCatalog()->
+                                    GetCounters().GetCsvMap();
+  } else if (attr == "user.repo_metainfo") {
+    if (!mount_point_->catalog_mgr()->manifest()) {
+      attribute_value = "Manifest not available";
+    } else {
+      shash::Any hash = mount_point_->catalog_mgr()->manifest()->meta_info();
+      int fd = mount_point_->fetcher()->
+                Fetch(hash, CacheManager::kSizeUnknown,
+                      "metainfo (" + metainfoHash.ToString() + ")",
+                      zlib::kZlibDefault, CacheManager::kTypeVolatile, "");
+      if (fd < 0) {
+        attribute_value = "Failed to open metadata file";
+      } else {
+        SafeReadToString(fd, &attribute_value);
```

Comment:
The file descriptor is not necessarily a system file descriptor. It is valid only in the context of the cache manager.  That means, we'd need to use cache manager methods to read from it and close it.